packages/web/pages/api/typedefs/article.ts

```diff
@@ -0,0 +1,174 @@
+import { ApolloError } from "apollo-server-micro";
+import { authenticated } from "./scalars";
+
+export const typeDef = `
+  extend type Query {
+    article(id: ID!): Article
+    articles: [Article]
+  }
+
+  extend type Mutation {
+    createdArticle(articleInput: ArticleCreateInput!): Article
+    updatedArticle(articleInput: ArticleUpdateInput): Article
+    
+    isFavourite(articleId: ID!): Boolean
+
+    moveArticle(id: ID!, parentId: ID): Boolean
+  }
+
+  input ArticleCreateInput {
+    title: String!
+    body: String
+    authorId: ID
+    parentId: ID
+  }
+
+  input ArticleUpdateInput {
+    id: ID
+    title: String
+    body: String
+    children: [ID]
```

Comment:
I think that in order to allow better updates children should be an Array of ArticleInput. 
This would allow some bulk updates in the future . 
I really like the idea that inputs should be as close as the real object. 
This way when we update or create we send the desired state of a resource including childrens and properties and the API figures out what it should to make all the changes needed.